include/solvers/vof.h

```diff
@@ -473,6 +546,37 @@ class VolumeOfFluid
   TrilinosWrappers::MPI::Vector  complete_system_rhs_phase_fraction;
   TrilinosWrappers::SparseMatrix mass_matrix_phase_fraction;
 
+  // Filtered phase fraction gradient (pfg) solution
+  TrilinosWrappers::MPI::Vector present_pfg_solution;
+  IndexSet                      locally_owned_dofs_pfg;
+  IndexSet                      locally_relevant_dofs_pfg;
+  AffineConstraints<double>     pfg_constraints;
+  TrilinosWrappers::MPI::Vector nodal_pfg_relevant;
+  TrilinosWrappers::MPI::Vector nodal_pfg_owned;
+
+  TrilinosWrappers::SparseMatrix system_matrix_pfg;
+  TrilinosWrappers::SparseMatrix complete_system_matrix_pfg;
```

Comment:
Why do you need two matrices? I don't understand

Reply:
It was a mistake. it is corrected.